### PR TITLE
Documented the checkDNS option of the Url validator

### PR DESCRIPTION
**Pull request by javiereguiluz**
| Q | A |
| --- | --- |
| Doc fix? | no |
| New docs? | yes |
| Applies to | 2.6+ |
| Fixed tickets | - |

### CODE REVIEW

File: reference/constraints/Url.rst [hunk 1]
```diff
@@ -84,13 +84,227 @@ message
...
+                $metadata->addPropertyConstraint('bioUrl', new Assert\Url(array(
+                    'protocols' => array('http', 'https', 'ftp'),
+                )));
+            }
+        }
+
+checkDNS
+~~~~~~~~
+
+**type**: ``Boolean`` **default**: ``false``
```
Comment (xabbuh) [inline reference/constraints/Url.rst]: boolean

**Comment by xabbuh** [comment]
@javiereguiluz I left just one minor comment.

Though unfortunately you need to do another rebase (the `payload` option was lost and the order of the code blocks needs to be fixed).

File: reference/constraints/Url.rst (hunk 1 above)
Comment (javiereguiluz) [inline reference/constraints/Url.rst]: Nice catch! Fixed. Thanks.

**Comment by javiereguiluz** [comment]
@xabbuh I've added the `payload` option removed by mistake (I don't know what could I do without your reviews!!! Thanks a lot). And the examples have also been sorted as annotation - yaml -xml - php.

**Comment by xabbuh** [comment]
Perfect! :+1:

File: reference/constraints/Url.rst [hunk 2]
```diff
@@ -84,13 +85,229 @@ message
...
+             protected $bioUrl;
+        }
+
+    .. code-block:: yaml
+
+        # src/Acme/BlogBundle/Resources/config/validation.yml
+        Acme\BlogBundle\Entity\Author:
+            properties:
+                bioUrl:
+                    - Url: ~
```
Comment (WouterJ) [inline reference/constraints/Url.rst]: the `~` is wrong (as the options are defined below)

File: reference/constraints/Url.rst [hunk 3]
```diff
@@ -84,13 +85,229 @@ message
...
+        use Symfony\Component\Validator\Constraints as Assert;
+
+        class Author
+        {
+            public static function loadValidatorMetadata(ClassMetadata $metadata)
+            {
+                $metadata->addPropertyConstraint('bioUrl', new Assert\Url(array(
+                    'message' => 'The url "{{ value }}" is not a valid url.',
+                    'protocols' => array('http', 'https'),
+                    'checkDNS' => true,
```
Comment (WouterJ) [inline reference/constraints/Url.rst]: please align `=>`

File: reference/constraints/Url.rst [hunk 4]
```diff
@@ -84,13 +85,229 @@ message
...
+             protected $bioUrl;
+        }
+
+    .. code-block:: yaml
+
+        # src/Acme/BlogBundle/Resources/config/validation.yml
+        Acme\BlogBundle\Entity\Author:
+            properties:
+                bioUrl:
+                    - Url: ~
```
Comment (WouterJ) [inline reference/constraints/Url.rst]: same here (and I prefer `- Url: { protocols: [http, https, ftp] }`)

File: reference/constraints/Url.rst [hunk 5]
```diff
@@ -84,13 +85,229 @@ message
...
```
Comment (WouterJ) [inline reference/constraints/Url.rst]: ...and here :)

**Comment by javiereguiluz** [comment]
I've fixed all the reported issues. Thanks.

**Comment by xabbuh** [comment]
Now for some reason the order of the code blocks in the first example changes again.

**Comment by javiereguiluz** [comment]
@xabbuh sorry for that! Fixed now. Thanks.

**Comment by xabbuh** [comment]
:+1:

**Comment by weaverryan** [comment]
Ok, got this guy merged - thanks Javier and @saro0h!

But, it got messy :). If you see any mistakes, please let me know:

* I merged most of this PR into the 2.3 branch - see sha: fea061ff052d9180fe2574c9aa7b9d8167ff0a29
* I re-added the payload option in 2.6
* I re-added the checkDNS stuff in 2.7 at sha: cc4c4489f8801993dde72148d9c37511ce76939d (with the `versionadded` tag

Cheers!

**Comment by xabbuh** [comment]
looks good :+1:

**Comment by javiereguiluz** [comment]
It looks good to me too. This complex merging process made me realize that I'm not really aware of the amount of work involved in merging complex PRs. Thanks Ryand and the rest of doc managers for your dedicated work!

**Comment by xabbuh** [comment]
@javiereguiluz Don't worry about that. This PR was a bit special as it involved changes for two different branches (unfortunately I didn't realise that during review).

**Comment by weaverryan** [comment]
It surprised me too - looked easy! :)

**Pull request closed without merging**